src/Symfony/Component/Form/Extension/Validator/Type/FormTypeValidatorExtension.php

```diff
@@ -29,10 +29,23 @@ class FormTypeValidatorExtension extends BaseValidatorExtension
 {
     private ValidatorInterface $validator;
     private ViolationMapper $violationMapper;
-    private bool $legacyErrorMessages;
 
-    public function __construct(ValidatorInterface $validator, bool $legacyErrorMessages = true, FormRendererInterface $formRenderer = null, TranslatorInterface $translator = null)
+    /**
+     * @param ValidatorInterface         $validator
+     * @param FormRendererInterface|null $formRenderer
+     * @param TranslatorInterface|null   $translator
+     */
+    public function __construct(ValidatorInterface $validator/* , FormRendererInterface $formRenderer = null, TranslatorInterface $translator = null*/)
     {
+        if (\func_num_args() > 3) {
+            trigger_deprecation('symfony/form', '6.3', 'The signature of constructor requires 3 arguments: "ValidatorInterface $validator, FormRendererInterface $formRenderer = null, TranslatorInterface $translator = null". Passing argument $legacyErrorMessages is deprecated.', __METHOD__);
+            $formRenderer = func_get_arg(2);
+            $translator = func_get_arg(3);
+        } else {
+            $formRenderer = func_get_arg(1);
+            $translator = func_get_arg(2);
+        }
+
```

Comment:
You'll have to check if the second argument is `bool` or use `instanceof` on the other two. This won't work if someone does `new FormTypeValidatorExtension($validator, true, $formRenderer)` as the number of arguments in that case is not greater than 3.